zaza/openstack/utilities/openstack.py

```diff
@@ -2693,6 +2694,28 @@ def upload_image_to_glance(glance, local_path, image_name, disk_format='qcow2',
     return image
 
 
+def get_app_names_for_charm(charm_name, model_name=None):
+    """Get all application names for a given charm in the model.
+
+    :returns: List of application names
+    :rtype: List[str]
+    """
+    status = juju_utils.get_full_juju_status(model_name=model_name)
+    result = []
+    try:
+        for app_name, app_data in status['applications'].items():
+            try:
+                if charm_name == app_data['charm-name']:
+                    result.append(app_name)
+            except KeyError:
+                # Older juju versions may not have the field "charm-name"
```

Comment:
I see references to "charm-name" as far back as juju 1.25 and it is definitely in 2.3, maybe this is overkill?

Reply:
I have seen juju status json without this field while coding probelius, then a later juju update introduced this field consistently. A bit overkill probably, but harmless I suppose.